### PR TITLE
159 log files and lines

### LINKED ISSUE #159: Log file names and line numbers

### PR DESCRIPTION
**Pull request by mpsijm**
Will close #159 

Commit hashes, file names and line numbers are now logged on diff elements.

Have fun reviewing!

### CODE REVIEW

File: src/octopeer-github/content/ElementSelectionBehaviour/DiffElementSelectionBehaviourFunctions.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Created by Maarten on 17-06-2016.
+ *
+ * Provide additional functions to help the processing of events on the diff.
+ * Is a utility class, so should be final but TypeScript doesn't support that.
+ */
+abstract class DiffElementSelectionBehaviourFunctions {
+
+    public static getFilenameFromDiffLine(line: JQuery) {
```
Comment (mdingjan) [inline src/octopeer-github/content/ElementSelectionBehaviour/DiffElementSelectionBehaviourFunctions.ts]: Some documentation plox :)

File: src/octopeer-github/content/ElementSelectionBehaviour/DiffElementSelectionBehaviourFunctions.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        return line.parent().parent().parent().parent().parent()
+            .children(".file-header").attr("data-path");
+    }
+
+    public static getCommitHashFromDiffLine(line: JQuery) {
```
Comment (mdingjan) [inline src/octopeer-github/content/ElementSelectionBehaviour/DiffElementSelectionBehaviourFunctions.ts]: Some documentation plox :)

**Comment by mdingjan** [comment]
Just a few minor comments; then this is ready for merging!

File: src/octopeer-github/main/Database/RESTApiDatabaseAdapter.ts [hunk 3]
```diff
@@ -64,12 +64,29 @@ class RESTApiDatabaseAdapter implements DatabaseAdaptable {
...
      * @param success       Callback, which is called once the call has succeeded.
      * @param failure       Callback, which is called once the call has failed.
      */
     public post(eventData: EventObject, success: Callback, failure: Callback): void {
-        this.postEvent(this.processEventObject(eventData), this.endPoints[eventData.type], success, failure);
+        let newSuccess = success;
+        if (eventData.type === "SemanticEvent") {
+            const data = <SemanticEvent>eventData.data;
+            if (data.commit_hash !== undefined && data.filename !== undefined && data.line_number !== undefined) {
+                // Override callback if a semanticEvent contains diff / line number information
```
Comment (thervh70) [inline src/octopeer-github/main/Database/RESTApiDatabaseAdapter.ts]: This is already documented in the method documentation right?

**Comment by mdingjan** [comment]
Thanks for resolving my feedback.

Ready for merging!

File: src/octopeer-github/test/content/ElementSelectionBehaviour/DiffElementSelectionBehaviourFunctionsTest.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    it("should return the right file name", function() {
+        loadFixtures("files-changed-tab.html");
+        expect(DiffElementSelectionBehaviourFunctions.getFilenameFromDiffLine($(".blob-num-addition"))).toEqual("addedFile.txt");
+    });
+
+    it("should return the right commit id", function() {
+        loadFixtures("files-changed-tab.html");
+        expect(DiffElementSelectionBehaviourFunctions.getCommitHashFromDiffLine($(".blob-num-addition")))
+            .toEqual("39efa2e95a60e240861dc85028d90f93aef7eabc");
```
Comment (Exclaminator) [inline src/octopeer-github/test/content/ElementSelectionBehaviour/DiffElementSelectionBehaviourFunctionsTest.ts]: Can this be put in an explanative constant?

**Comment by Exclaminator** [comment]
If my comment is resolved this can be merged in my opinion.

**Comment by thervh70** [comment]
2 reviews done! Will merge.

**Pull request merged**